lib/src/bindings/signatures.dart

```diff
@@ -159,6 +159,14 @@ typedef obx_query_visit_native_t = Int32 Function(Pointer<Void> query,
 typedef obx_query_visit_dart_t = int Function(Pointer<Void> query,
     Pointer<NativeFunction<obx_data_visitor_native_t>> visitor, Pointer<Void> user_data, int offset, int limit);
 
+// observers
+
+typedef obx_observer_t<U extends NativeType, T extends NativeType> = U Function(Pointer<Void> user_data, Pointer<Uint32> entity_id, T type_ids_count);
+typedef obx_observer_single_type_t<U extends NativeType> = U Function(Pointer<Void> user_data);
+typedef obx_observe_t = Pointer<Void> Function(Pointer<Void> store, Pointer<NativeFunction<obx_observer_t<Void, Uint32>>> callback, Pointer<Void> user_data);
+typedef obx_observe_single_type_t<T> = Pointer<Void> Function(Pointer<Void> store, T entity_id, Pointer<NativeFunction<obx_observer_single_type_t<Void>>> callback, Pointer<Void> user_data);
+typedef obx_observer_close_t<U extends NativeType> = U Function(Pointer<Void> observer);
```

Comment:
As said most of these use AFAIC tell unnecessary type args. Is there any reason for this complication?
E.g.
```
typedef obx_observer_t = Void Function(Pointer<Void> user_data, Pointer<Uint32> entity_id, Pointer<Uint32> type_ids_count);
```
should be enough?

Reply:
sorry, real life got in the way, I made the changes!

Reply:
Thanks, I'll have a look once I can make some time.